ui/components/multichain/network-manager/components/total-balance/total-balance.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useCallback, memo } from 'react';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import {
+  Icon,
+  IconName,
+  IconSize,
+  Popover,
+  PopoverPosition,
+} from '../../../../component-library';
+import { Box } from '../../../../component-library/box';
+import { Text } from '../../../../component-library/text';
+
+type TotalBalanceProps = {
+  /**
+   * The total balance amount to display
+   */
+  totalAmount?: string;
+  /**
+   * Number of networks loaded out of total networks
+   */
+  networksLoaded?: number;
+  /**
+   * Total number of networks
+   */
+  totalNetworks?: number;
+};
+
+/**
+ * TotalBalance component
+ *
+ * Displays the total balance amount across networks with a loading indicator.
+ * Shows how many networks have been loaded out of the total networks.
+ * Includes a hover tooltip with network loading information.
+ *
+ * @param props - Component props
+ * @returns TotalBalance component
+ */
+export const TotalBalance = memo(
+  ({
+    totalAmount = '$12.00',
+    networksLoaded = 1,
+    totalNetworks = 8,
+  }: TotalBalanceProps) => {
+    const t = useI18nContext();
+    const [isOpen, setIsOpen] = useState(false);
+    const [referenceElement, setReferenceElement] =
+      useState<HTMLDivElement | null>(null);
+
+    const handleMouseEnter = useCallback(() => {
+      setIsOpen(true);
+    }, []);
+
+    const handleMouseLeave = useCallback(() => {
+      setIsOpen(false);
+    }, []);
+
+    const setBoxRef = useCallback((ref: HTMLDivElement | null) => {
+      setReferenceElement(ref);
+    }, []);
+
+    return (
+      <Box
+        display={Display.Flex}
+        alignItems={AlignItems.center}
+        flexDirection={FlexDirection.Row}
+        gap={3}
+        onMouseLeave={handleMouseLeave}
+        style={{
+          position: 'relative',
+        }}
+      >
+        <Text
+          variant={TextVariant.bodyMdMedium}
+          color={TextColor.textAlternative}
+        >
+          {t('total')} {totalAmount}
```

Comment:
This component looks good to me, the only suggestion is that we could use the aggregated balance component for total amount cc @gambinish